bin/find-never-defined-css-variables/find-never-defined-css-variables.ts

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env tsx
+/* eslint-disable no-console */
+import { readFileSync } from "fs";
+
+import { glob } from "glob";
+
+/**
+ * This script finds css variables that are used but *never* defined in our codebase.
+ * Its goal isn't to find usages of variables in contexts where that variable is not defined,
+ * as that would be a lot harder.
+ * Its goal is to prevent typos and wrong usages created by conflicts and backports.
+ * The most common thing this should catch and prevent is backporting a fix or feature that uses a css variable
+ * that doesn't exist in the release branch.
+ */
+
+const shouldWhiteList = (_variable: string) => {
+  // Use this for whitelisting some variables if we know they're defined by a 3rd party, such as mantine
+  // Note: for be sure to not backport changes to this function that would whitelist variables not on the release branch
+  return false;
+};
+
+interface UsageMap {
+  // variable : files
+  [variable: string]: string[];
+}
+
+const findFiles = (): string[] => {
+  // TODO: only frontend and /enterprise?
+  return glob.sync("**/*.{css,module.css,js,jsx,ts,tsx}", {
+    ignore: ["**/node_modules/**", "**/dist/**", "**/target/**", "bin/**"],
```

Comment:
Let's also ignore `/cypress` directory.

We may have some .css/.js there

![image](https://github.com/user-attachments/assets/b506020f-2442-42a5-90dc-2e696ab66685)